clients/client/src/client.js

```diff
@@ -182,8 +202,13 @@ exports.createClient = function(reference, name) {
       serviceName,
       serviceVersion: 'v1',
     }, _defaultOptions);
+    assert(this._options.rootUrl, 'Must provide a rootUrl'); // We always assert this even with service discovery
+
+    const serviceDiscoveryScheme = options.serviceDiscoveryScheme || DEFAULT_SERVICE_DISCOVERY_SCHEME;
 
-    assert(this._options.rootUrl, 'Must provide a rootUrl');
+    if (serviceDiscoveryScheme === 'k8s-dns') {
```

Comment:
How do we know what this deployment is other than being told that in the rootUrl? basically that makes sense but what to compare it to here?

Reply:
I was thinking of comparing it to `$TASKCLUSTER_ROOT_URL`, but maybe that's a bad idea since we don't look at environment variables anywhere else.  ¯\_(ツ)_/¯